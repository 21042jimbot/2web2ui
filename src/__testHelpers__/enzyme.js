export const findButton = (wrapper, buttonLabel) =>
  wrapper.findWhere(
    (node) => node.type() === 'button' && new RegExp(buttonLabel).test(node.text())
  );
