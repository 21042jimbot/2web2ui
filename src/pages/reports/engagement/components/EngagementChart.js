import asyncComponent from 'src/components/asyncComponent/asyncComponent';
import PanelLoading from 'src/components/panelLoading/PanelLoading';

export default asyncComponent(() => import('./async/EngagementChart'), PanelLoading);
