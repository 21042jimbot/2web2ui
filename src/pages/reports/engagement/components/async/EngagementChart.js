import React from 'react';
import { Bar, BarChart, CartesianGrid, LabelList, ResponsiveContainer, XAxis, YAxis } from 'recharts';

import styles from './EngagementChart.module.scss';

export default function EngagementChart({ data = []}) {
  // Increased top margin to vertically center the chart
  const margins = { bottom: 5, left: 5, right: 5, top: 25 };
  // Pad between tallest bar and top of the chart to provide room for the LabelList
  const topPadding = { top: 60 };

  return (
    <ResponsiveContainer className={styles.EngagementChart} height={350} width="100%">
      <BarChart barCategoryGap="35%" data={data} margin={margins}>
        <CartesianGrid strokeDasharray="4 1" vertical={false} />
        <XAxis dataKey="name" tickLine={false} />
        <YAxis padding={topPadding} tickLine={false} />
        <Bar dataKey="value" fill="#37aadc" isAnimationAction={false}>
          <LabelList fill="#55555a" position="top" />
        </Bar>
      </BarChart>
    </ResponsiveContainer>
  );
}
