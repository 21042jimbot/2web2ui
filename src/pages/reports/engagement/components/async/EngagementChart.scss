@import '~@sparkpost/matchbox/src/styles/config.scss';

.sp-engagement-chart {
  position: relative; // for the completion rate

  // hide the top grid line
  .recharts-wrapper .recharts-cartesian-grid-horizontal line:last-child {
    stroke-opacity: 0;
  }

  // similar to match summary line  chart
  .recharts-text.recharts-cartesian-axis-tick-value {
    fill: color(gray, 5);
    font-size: rem(13);
    font-weight: 500;
  }

  .recharts-cartesian-axis-line {
    stroke: color(gray, 7);
  }

  .recharts-text.recharts-cartesian-axis-tick-value {
    fill: color(gray, 5);
    font-size: 12px;
    font-weight: 500;
  }

  .xAxis .recharts-text.recharts-cartesian-axis-tick-value {
    transform: translate(0, 6px);
  }

  .recharts-cartesian-grid-horizontal line {
    stroke: color(gray, 8);
  }
}
