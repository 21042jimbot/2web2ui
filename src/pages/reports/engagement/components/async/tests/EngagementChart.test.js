import React from 'react';
import { shallow } from 'enzyme';
import EngagementChart from '../EngagementChart';

it('renders engagement chart', () => {
  const wrapper = shallow(<EngagementChart />);
  expect(wrapper).toMatchSnapshot();
});
