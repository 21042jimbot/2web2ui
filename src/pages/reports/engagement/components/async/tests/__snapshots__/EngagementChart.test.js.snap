// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders engagement chart 1`] = `
<ResponsiveContainer
  debounce={0}
  height={350}
  width="100%"
>
  <BarChart
    barCategoryGap="35%"
    barGap={4}
    data={Array []}
    layout="horizontal"
    margin={
      Object {
        "bottom": 5,
        "left": 5,
        "right": 5,
        "top": 25,
      }
    }
    reverseStackOrder={false}
    stackOffset="none"
  >
    <CartesianGrid
      fill="none"
      horizontal={true}
      horizontalPoints={Array []}
      stroke="#ccc"
      strokeDasharray="4 1"
      vertical={false}
      verticalPoints={Array []}
    />
    <XAxis
      allowDataOverflow={false}
      allowDecimals={true}
      dataKey="name"
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={30}
      hide={false}
      mirror={false}
      orientation="bottom"
      padding={
        Object {
          "left": 0,
          "right": 0,
        }
      }
      reversed={false}
      scale="auto"
      tickCount={5}
      tickLine={false}
      type="category"
      width={0}
      xAxisId={0}
    />
    <YAxis
      allowDataOverflow={false}
      allowDecimals={true}
      domain={
        Array [
          0,
          "auto",
        ]
      }
      height={0}
      hide={false}
      mirror={false}
      orientation="left"
      padding={
        Object {
          "top": 60,
        }
      }
      reversed={false}
      scale="auto"
      tickCount={5}
      tickLine={false}
      type="number"
      width={60}
      yAxisId={0}
    />
    <Bar
      animationBegin={0}
      animationDuration={400}
      animationEasing="ease"
      data={Array []}
      dataKey="value"
      fill="#37aadc"
      hide={false}
      isAnimationAction={false}
      isAnimationActive={true}
      layout="vertical"
      legendType="rect"
      minPointSize={0}
      onAnimationEnd={[Function]}
      onAnimationStart={[Function]}
      xAxisId={0}
      yAxisId={0}
    >
      <LabelList
        fill="#55555a"
        position="top"
        valueAccessor={[Function]}
      />
    </Bar>
  </BarChart>
</ResponsiveContainer>
`;
